edgelet/iotedged/src/lib.rs

```diff
@@ -317,39 +320,39 @@ impl Main {
                         )))
                     });
 
-                let provisioning = tokio_runtime.block_on(provision_fut);
-
-                provisioning.and_then(move |prov_result| {
-                    prov_result
-                        .credentials()
-                        .map_or_else(
-                            || {
-                                info!("Credentials are expected to be populated for external provisioning.");
-                                Err(Error::from(ErrorKind::Initialize(
-                                    InitializeErrorReason::ExternalProvisioningClient,
-                                )))
-                            },
-                            |credentials| {
-                                if let AuthType::SymmetricKey(symmetric_key) = credentials.auth_type() {
-                                    if let Some(key) = symmetric_key.key() {
-                                        let (derived_key_store, memory_key) = external_provision_payload(key);
-                                        start_edgelet!(derived_key_store, prov_result, memory_key, runtime);
-                                        Ok(())
-                                    }
-                                    else {
-                                        let (derived_key_store, tpm_key) = external_provision_tpm(hsm_lock.clone())?;
-                                        start_edgelet!(derived_key_store, prov_result, tpm_key, runtime);
-                                        Ok(())
-                                    }
-                                }
-                                else {
-                                    info!("Unexpected auth type. Only symmetric keys are expected");
-                                    Err(Error::from(ErrorKind::Initialize(
-                                        InitializeErrorReason::ExternalProvisioningClient,
-                                    )))
-                                }
-                            })
-                })?;
+                let prov_result = tokio_runtime.block_on(provision_fut)?;
+
+                prov_result.credentials().map_or_else(
```

Comment:
As I said in the previous comment, you don't need a `map_or_else` here either.

Reply:
I see, updated.